devito/tools/abc.py

```diff
@@ -177,6 +177,9 @@ def func(self):
     def _evaluate_args(self):
         return [Evaluable._evaluate_maybe_nested(i) for i in self.args]
 
+    def _eval_at(self, func):
```

Comment:
this is a hack, and needs to go: there is no reason _eval_at should appear in this class

Reply:
It's not a hack, all Evaluable object have to have an `_eval_at`.

Reply:
The concept of "evaluating at", subsuming there's a point/object we're referring to, makes no sense, _unless_ we move Evaluable to `types` and _change_ its semantics

Reply:
May be able to remove it will look into I guess